lib/Doctrine/DBAL/Driver/PDOConnection.php

```diff
@@ -84,17 +83,30 @@ public function query(string $sql) : ResultStatement
     /**
      * {@inheritdoc}
      */
-    public function quote($input, $type = ParameterType::STRING)
+    public function quote(string $input) : string
     {
-        return $this->connection->quote($input, $type);
+        return $this->connection->quote($input);
     }
 
     /**
      * {@inheritdoc}
      */
-    public function lastInsertId($name = null)
+    public function lastInsertId(?string $name = null) : string
     {
-        return $this->connection->lastInsertId($name);
+        try {
+            $lastInsertId = $this->connection->lastInsertId($name);
+        } catch (\PDOException $e) {
+            throw new PDOException($e);
+        }
+
+        if ($lastInsertId === '0' || $lastInsertId === '') {
```

Comment:
Is this needed since different drivers return different empty values? If so, please add a comment to the code.

Reply:
It is indeed. Added a comment.